jetbrains-gateway/src/software/aws/toolkits/jetbrains/gateway/GatewayDeprecationNotice.kt

```diff
@@ -0,0 +1,35 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.gateway
+
+import com.intellij.ide.AppLifecycleListener
+import com.intellij.notification.NotificationGroupManager
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.application.ApplicationInfo
+import com.intellij.openapi.application.ApplicationNamesInfo
+import software.aws.toolkits.resources.message
+
+class GatewayDeprecationNotice : AppLifecycleListener {
```

Comment:
should we share logic with the one with the regular IDEs?

Reply:
We can't reuse the logic that we do for the regular IDEs since the functions and classes used eg StartupActivity.runActivity and AppLifecycleListener.welcomeScreenDisplayed() are different and not the first one cannot be applied in Gateway. Most of the other logic used inside for comparing min version stays the same...